apps/gitness/src/components-v2/file-content-viewer.tsx

```diff
@@ -89,6 +89,23 @@ export default function FileContentViewer({ repoContent }: FileContentViewerProp
     navigate(`edit/${fullGitRef}/~/${fullResourcePath}`)
   }
 
+  const CodeView = () => {
+    return useMemo(
+      () => (
+        <CodeEditor
+          language={language}
+          codeRevision={{ code: fileContent }}
+          onCodeRevisionChange={() => {}}
+          themeConfig={themeConfig}
+          options={{
+            readOnly: true
+          }}
+        />
+      ),
+      [language, fileContent, themeConfig]
+    )
+  }
```

Comment:
This is a component that's entirely within another component and will be redefined on every render. Let's move this out to its own thing, or inline it where needed.

Reply:
this is required for Monaco Code editor to update when fileContent changes from repo-code page itself - ex. commiting changes to current file on the currentBranch itself.

Reply:
Having a component defined within a component means that it will be redefined on every render and React will lose track and have to rerender in its entirety every time. Have a look here for a good explanation https://youtu.be/2sAdzy90GtE

Reply:
right I understand, what I meant was that this unmount and mount was the only way I could get monaco CodeEditor to update and show changed after a file was changed from edit mode to view mode.
will sync with @srdjan-harness to see if it can be fixed from the editor side

Reply:
reverting this as @srdjan-harness fixed the changes from Editor side in this PR - https://github.com/harness/canary/pull/585